runtime/doc/treesitter.txt

```diff
@@ -1605,14 +1608,20 @@ LanguageTree:parse({range})                             *LanguageTree:parse()*
     if {range} is `true`).
 
     Parameters: ~
-      • {range}  (`boolean|Range?`) Parse this range in the parser's source.
-                 Set to `true` to run a complete parse of the source (Note:
-                 Can be slow!) Set to `false|nil` to only parse regions with
-                 empty ranges (typically only the root tree without
-                 injections).
+      • {range}     (`boolean|Range?`) Parse this range in the parser's
+                    source. Set to `true` to run a complete parse of the
+                    source (Note: Can be slow!) Set to `false|nil` to only
+                    parse regions with empty ranges (typically only the root
+                    tree without injections).
+      • {on_parse}  (`fun(trees: TSTrees, timeout_reached: boolean)?`)
+                    Function invoked when parsing completes. When provided,
+                    parsing will run asynchronously. The function is passed
+                    the list of trees returned by the parse, as well as a
+                    boolean indicating whether or not the parse timed out,
+                    determined by 'redrawtime'.
```

Comment:
Also at `:help 'redrawtime'`, would you mind adding `... and async |LanguageTree:parse()|`.